src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -1081,6 +1092,936 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
+                // t0 = *  HWINTRINSIC simd16 float Shuffle
+
+                assert(comp->compIsaSupportedDebugOnly(InstructionSet_SSE));
+
+                node->gtOp1 = op1;
+
+                LIR::Use t1Use(BlockRange(), &node->gtOp1, node);
+                ReplaceWithLclVar(t1Use);
+
+                node->gtOp2 = comp->gtClone(node->gtOp1);
+                BlockRange().InsertBefore(node, node->gtOp2);
+
+                idx = comp->gtNewIconNode(0x00, TYP_INT);
+                BlockRange().InsertBefore(node, idx);
+
+                node->gtOp1 = comp->gtNewArgList(node->gtOp1, node->gtOp2, idx);
+                node->gtOp2 = nullptr;
+
+                node->gtHWIntrinsicId = NI_SSE_Shuffle;
+                break;
+            }
+
+            case TYP_DOUBLE:
+            {
+                if (comp->compOpportunisticallyDependsOn(InstructionSet_SSE3))
+                {
+                    //      /--*  <value>
+                    // t1 = *  HWINTRINSIC simd16 double CreateScalarUnsafe
+                    //      /--*  t1     simd16
+                    // t0 = *  HWINTRINSIC simd16 double MoveAndDuplicate
+
+                    node->gtOp1 = op1;
+                    node->gtOp2 = nullptr;
+
+                    node->gtHWIntrinsicId = NI_SSE3_MoveAndDuplicate;
+                    break;
+                }
+
+                assert(comp->compIsaSupportedDebugOnly(InstructionSet_SSE2));
+
+                //      /--*  <value>
+                // t1 = *  HWINTRINSIC simd16 double CreateScalarUnsafe
+                //      *  STORE_LCL_VAR simd16<Vector128`1>
+                // t2 =    LCL_VAR   simd16<Vector128`1>
+                // t3 =    LCL_VAR   simd16<Vector128`1>
+                //      /--*  t2     simd16
+                //      +--*  t3     simd16
+                // t0 = *  HWINTRINSIC simd16 float MoveLowToHigh
+
+                node->gtOp1 = op1;
+
+                LIR::Use op1Use(BlockRange(), &node->gtOp1, node);
+                ReplaceWithLclVar(op1Use);
+
+                node->gtOp2 = comp->gtClone(node->gtOp1);
+                BlockRange().InsertBefore(node, node->gtOp2);
+
+                node->gtHWIntrinsicId = NI_SSE_MoveLowToHigh;
+                node->gtSIMDBaseType  = TYP_FLOAT;
+
+                break;
+            }
+
+            default:
+            {
+                unreached();
+            }
+        }
+
+        return;
+    }
+
+    // We have the following:
+    //
+    // /--*  <valueFirst>
+    // +--*  <value...>
+    // +--*  <valueLast>
+    // *  HWINTRINSIC simd16 T Create
+
+    if (intrinsicId == NI_Vector256_Create)
+    {
+        assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+
+        // We will be constructing the following parts:
+        //
+        // t2 =    CNS_INT   int    0
+        //      /--*  t?     simd32
+        //      +--*  t?     simd16
+        //      +--*  t?     int
+        // t0 = *  HWINTRINSIC simd32 T InsertVector128
```

Comment:
In addition to clarifying what the operands represent, it would also be good to add some additional comments to describe what's being done.

Reply:
Added comments giving the "managed equivalent" that the intrinsic is being lowered to